src/components/oInput.js

```diff
@@ -0,0 +1,227 @@
+  if (!(this instanceof oInput)) {
+    return new oInput(typeOrConfig);
+  }
+
+  this.element = o('input');
+  this.input = this.element.element;
+  this.dbSet = false;
+  this.dbObject = null;
+  this.dbKey = null;
+  this.index = null;
+
+  if(typeof typeOrConfig === 'string') {
+    this.type(typeOrConfig);
+  }
+
+  //TODO: initConfigService
+  if(typeof typeOrConfig === 'object') {
+    // this.attr(typeOrConfig);
+  }
+}
+
+oInput.prototype.attr = function(attrs) {
+  this.element.attr(attrs);
+  return this;
+}
+
+oInput.prototype.type = function(type) {
+  this.element.attr({
+    type
+  });
+
+  return this;
+}
+
+oInput.prototype.name = function(name) {
+  if(typeof name === 'undefined') return this;//nie wiem czy sprawdzać też pusty string + inne walidacje
+  this.element.attr({
+    name
+  });
+  return this;
+}
+
+oInput.prototype.required = function(required) {
+  if(typeof required !== 'boolean') return this;
+
+  this.element.attr({
+    required
+  });
+  return this;
+}
+
+oInput.prototype.disabled = function(required) {
+  if(typeof disabled !== 'boolean') return this;
+
+  this.element.attr({
+    disabled
+  });
+  return this;
+}
+
+oInput.prototype.readOnly = function(readonly) {
+  if(typeof readonly !== 'boolean') return this;
+
+  this.element.attr({
+    readonly
+  });
+  return this;
+}
+
+oInput.prototype.id = function (id) {
+  if(typeof id === 'undefined') return this;//nie wiem czy sprawdzać też pusty string + inne walidacje
+
+  this.element.id(id);
+  return this;
+}
+
+oInput.prototype.onChange = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'change',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onKeyUp = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'keyup',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onKeyDown = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'keydown',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onFocus = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'focusin',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onFocusOut = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'focusout',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.value = function(value) {
+  this.element.setValue(value);
+  return this;
+}
+
+oInput.prototype.getValue = function() {
+  return this.input.value;
+}
+oInput.prototype.placeholder = function(placeholder) {
+  this.element.attr({
+    placeholder
+  });
+  return this;
+}
+
+oInput.prototype.classList = function(classList) {
+  this.element.classList(classList);
+  return this;
+}
+
+oInput.prototype.index = function(index) {
```

Comment:
Jak ręcznie ustawimy index to do czego to może się przydać?

Reply:
Chodzi o dostanie się do odpowiedniego elementu w tablicy w naszym db/store.

Reply:
zmieniam na dbIndex